### PR TITLE
LPS-144266 Test Autom. to be able to Replace/delete 'Hide A/B testing panel' button | AB Test

### PR DESCRIPTION
**Pull request by mateusralv**
JIRA Ticket:
[LPS-144266](https://issues.liferay.com/browse/LPS-144266)

Motivation:
"Replace / delete "Hide A/B testing panel" button"

### CODE REVIEW

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
ci:test:tango-analytics-cloud

**Comment by mateusralv** [comment]
Hello @Tim-Cao, could you please review this PR?

**Comment by Tim-Cao** [comment]
Just start reviewing

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase [hunk 1]
```diff
@@ -7,7 +7,8 @@ definition {
 
 	setUp {
 		task ("Go to DXP portal, login and create site") {
-			TestCase.setUpPortalInstance();
+
+			//TestCase.setUpPortalInstance();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase:11]: Uncomment

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase [hunk 2]
```diff
@@ -70,6 +71,77 @@ definition {
...
+
+			Click(locator1 = "AnalyticsCloudConnection#CONNECT_ANALYTICS_CLOUD_BUTTON");
+
+			AssertTextEquals(
+				locator1 = "AnalyticsCloudConnection#ANALYTICS_CLOUD_CONNECTED_CONFIRMATION_MESSAGE",
+				value1 = "Your DXP instance is connected to Analytics Cloud.");
+		}
+
+		task ("Go to the content page created") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/test-site-name/content-page");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase:119]:
```suggestion
			ContentPagesNavigator.openViewContentPage(
				pageName = "Content Page",
				siteName = "Test Site Name");
```

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path [hunk 3]
```diff
@@ -245,6 +245,11 @@
 	<td>//button[contains(@class,'btn-primary')and contains(text(),'Set Element as Click Target')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>SIDEBAR_MESSAGE</td>
+	<td>//div[contains(@class,"text-center")]//p[contains(text(),'${key_message}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path:250]: `//div[contains(@class,'sidebar-body')]//p[contains(text(),'${key_message}')]`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase [hunk 4]
```diff
@@ -70,6 +71,77 @@ definition {
...
+		}
+
+		task ("Open and assert that Ab Test Panel displays instructions to sync to AC and do not 'Don't show me this again' button and the message is not displayed") {
+			ABTest.openABTestPanel();
+
+			AssertElementPresent(
+				key_title = "Sync to Liferay Analytics Cloud",
+				locator1 = "Header#H4_TITLE");
+
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase:129]: AssertVisible

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase [hunk 5]
```diff
@@ -70,6 +71,77 @@ definition {
...
+		}
+
+		task ("Go to the content page created") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/test-site-name/content-page");
+		}
+
+		task ("Open and assert that Ab Test Panel displays instructions to sync to AC and do not 'Don't show me this again' button and the message is not displayed") {
+			ABTest.openABTestPanel();
+
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase:125]: AssertVisible

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase [hunk 6]
```diff
@@ -509,6 +509,28 @@ definition {
...
+	@priority = "5"
+	test CheckABTestPanelInfo {
+		property portal.acceptance = "true";
+
+		task ("Open AB Test Panel") {
+			ABTest.openABTestPanel();
+		}
+
+		task ("Assert that the message and the button for creating the AB Test are present") {
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:522]: AssertVisible

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase [hunk 7]
```diff
@@ -509,6 +509,28 @@ definition {
...
+				key_title = "No active tests were found for the selected experience.",
+				locator1 = "Header#H4_TITLE");
+
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:526]: AssertVisible

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase [hunk 8]
```diff
@@ -70,6 +71,77 @@ definition {
...
+			Click(locator1 = "AnalyticsCloudConnection#DXP_TOKEN_INPUT");
+
+			Paste(locator1 = "AnalyticsCloudConnection#DXP_TOKEN_INPUT");
+
+			Click(locator1 = "AnalyticsCloudConnection#CONNECT_ANALYTICS_CLOUD_BUTTON");
+
+			AssertTextEquals(
+				locator1 = "AnalyticsCloudConnection#ANALYTICS_CLOUD_CONNECTED_CONFIRMATION_MESSAGE",
+				value1 = "Your DXP instance is connected to Analytics Cloud.");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase:116]: Add a new site that is not synced to AC by default

**Review by Tim-Cao — COMMENTED** [review]
Hi @mateusralv, could you update according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase (hunk 8 above)
Comment (mateusralv) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTestWithoutACConnection.testcase:116]: great idea, thanks. I moved to the other file, as this new way the connection with the AC can be done normally.

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
Hello @Tim-Cao, could you please review my corrections ?

**Comment by Tim-Cao** [comment]
Good job @mateusralv, see changes https://github.com/liferay-tango/liferay-portal/pull/2023/commits/df691fe1892082ad9f182b0556a403bdf0efe49d

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**